### ISSUE TITLE
Should some rogue MongoDB migrations be rolled back?

### ISSUE THREAD

**Issue by turbomam** [microbiomedata/nmdc-schema#1303]
TL;DR: I am going to describe some migration-like `id` changes that were applied to MongoDB in the past. I am asking the following people to help me decide if an attempt should be made to roll back the changes: @Michal-Babins @mbthornton-lbl @picowatt @corilo @SamuelPurvine ?

cc @aclum @eecavanna 

"Rogue MongoDB migrations" means migrations that I (@turbomam) applied without keeping good logs and possibly without consulting other team members enough. I believe this is limited to coercing values in MongoDB that are supposed to be CURIes into CURIes by
- Removing illegal characters like white-spaces, carriage returns, etc.
- Adding a "nmdc:" prefix where prefixes were missing

## Elaboration

If I overstepped here, it was because I hadn't done a good job of thinking about the requirements for each step in the following process

1. download data from MongoDB and save it as a LinkML-style YAML file, by wrapping each collection in a nmdc-schema `Database` slot like `biosample-set`
2. applying agreed-upon migrations to the data file
3. validating the migrated data against a schema that expects to see those migrations but is very tolerant when it comes to legacy `id`s
4. making additional changes to the migrated data so that it can be converted to RDF (and loaded into GraphDB)

I think my main mistake was applying CURIes repairs as part of 2. instead of doing it in 4., such that the repaired `id`s were loaded back into MongoDB during the first few migrations that I did. I don't believe this combining of concerns has continued since @eecavanna has been leading the migration process.

see also
- https://github.com/microbiomedata/nmdc-schema/issues/1130#issuecomment-1796134109

**Comment by turbomam** [comment]
Since this rouge id repair happened before the Napa MongoDB was created, @aclum and others are inclined to skip a roll=back

would be helpful for @turbomam to find notes about when this happened

**Comment by turbomam** [comment]
@aclum @eecavanna here's an example of an entity that is rouge-re-ided in the nmdc-schema repo's `make make-rdf`, but not re-ided by the migration implementation from @eecavanna, @brynnz22 and @Shalsh23  

https://api-dev.microbiomedata.org/nmdcschema/ids/60e6fc9e49af89de1ba8999f

```json
{
  "id": "60e6fc9e49af89de1ba8999f",
  "name": "500088_1781_100336_MSGFjobs_MASIC_resultant.tsv",
  "description": "Aggregation of analysis tools{MSGFplus, MASIC} results",
  "file_size_bytes": 17527779,
  "type": "nmdc:DataObject",
  "url": "https://nmdcdemo.emsl.pnnl.gov/proteomics/500088_1781_100336_MSGFjobs_MASIC_resultant.tsv",
  "data_object_type": "Unfiltered Metaproteomics Results"
}
```

That will have the "generic:" prefix asserted in the next RDF dump I prepare for the Napa squad's data

**Comment by SamuelPurvine** [comment]
FWIW, this example above is from two generations ago of running the metaproteomics workflow, wherein we were searching genome annotations from Shane's first iteration of his workflow (which is the source of the "1781_100336" part of the name). The file that this refers to isn't even on the proteomics data repository (mdcdemo.emsl.pnnl.gov) anymore due to being superseded by the second metagenome annotation iteration that used short hash codes for ids, which in turn will be superseded by the current re-iding efforts. This would be an example of something we shouldn't even track anymore moving forward, and I would guess there will be many many more such records, especially when PNNL re-runs its workflow once Shane has re-id'd the metagenomes. I would be happy to help identify things that wholesale should just be removed from ever being tracked again, and I believe there is effort to track what is currently in the system in case anyone ever asks where the records went.

**Comment by aclum** [comment]
No longer relevant after re-iding.
